Tweak the reiser4 patch to apply when we use the splash patch

--- grub-0.97-reiser4-20050808-gentoo.patch
+++ grub-0.97-reiser4-20050808-gentoo.patch
@@ -455,3 +455,3 @@
  	fsys_vstafs.c fsys_xfs.c gunzip.c md5.c serial.c stage2.c \
- 	terminfo.c tparm.c
+ 	terminfo.c tparm.c graphics.c
  libgrub_a_CFLAGS = $(GRUB_CFLAGS) -I$(top_srcdir)/lib \
@@ -499,3 +499,3 @@
 @@ -79,7 +89,7 @@ else
- HERCULES_FLAGS =
+ GRAPHICS_FLAGS =
  endif
@@ -504,6 +504,6 @@
 +STAGE2_COMPILE = $(STAGE2_CFLAGS) -fno-builtin \
- 	$(NETBOOT_FLAGS) $(SERIAL_FLAGS) $(HERCULES_FLAGS)
+ 	$(NETBOOT_FLAGS) $(SERIAL_FLAGS) $(HERCULES_FLAGS) $(GRAPHICS_FLAGS)
  
  STAGE1_5_LINK = -nostdlib -Wl,-N -Wl,-Ttext -Wl,2000
-@@ -89,14 +99,17 @@ STAGE1_5_COMPILE = $(STAGE2_COMPILE) -DN
+@@ -89,15 +99,17 @@ STAGE1_5_COMPILE = $(STAGE2_COMPILE) -DN
  pre_stage2_exec_SOURCES = asm.S bios.c boot.c builtins.c char_io.c \
@@ -512,6 +512,7 @@
 -	fsys_reiserfs.c fsys_ufs2.c fsys_vstafs.c fsys_xfs.c gunzip.c \
--	hercules.c md5.c serial.c smp-imps.c stage2.c terminfo.c tparm.c
+-	hercules.c md5.c serial.c smp-imps.c stage2.c terminfo.c tparm.c \
+-	graphics.c
 +	fsys_reiserfs.c fsys_reiser4.c fsys_ufs2.c fsys_vstafs.c \
 +	fsys_xfs.c gunzip.c hercules.c md5.c serial.c smp-imps.c \
-+	stage2.c terminfo.c tparm.c
++	stage2.c terminfo.c tparm.c graphics.c
  pre_stage2_exec_CFLAGS = $(STAGE2_COMPILE) $(FSYS_CFLAGS)
